### PR TITLE
cuda support for new linear algebra operators

### PR DESCRIPTION
**Pull request by asmushetzel**
Add cuda support for syrk/gelqf linear algebra operators which was missing so far. 
Also fixes an issue in linalg_sumlogdiag that can lead to NAN values generated during backward pass.

### CODE REVIEW

**Comment by asmushetzel** [comment]
please do not merge immediately. still working on getting all tests through.

**Comment by asmushetzel** [comment]
Generally, all is there. Just that this code uses a cusolver-function that has been added with Cuda 8.0 and MxNet does builds on Cuda7.5. So I have to put in a workaround for disabling one operator cleanly on Cuda7.5 and also ensure that no GPU-tests are run on 7.5 for this operator. 
Will put in a comment when this is done.

**Comment by asmushetzel** [comment]
So this is done now. From my point of view, it can be merged.

File: src/operator/linalg_impl.h [hunk 1]
```diff
@@ -519,7 +519,7 @@ void linalg_trmm<gpu, DType>(const Tensor<gpu, 2, DType>& A, const Tensor<gpu, 2
                             (rightside ? CUBLAS_SIDE_LEFT : CUBLAS_SIDE_RIGHT), \
                             (lower ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER), \
                             (transpose ? CUBLAS_OP_T : CUBLAS_OP_N), \
-                            CUBLAS_DIAG_NON_UNIT, B.size(0), B.size(1), &alpha, \
+                            CUBLAS_DIAG_NON_UNIT, B.size(1), B.size(0), &alpha, \
```
Comment (piiswrong) [inline src/operator/linalg_impl.h]: was this a bug?

Comment (asmushetzel) [reply src/operator/linalg_impl.h]: Yes. Found it during testing the other operators

**Pull request merged**